DOC-10466 - Add Information on Minimum Memory Resident Ratio

sarahlwelton opened the pull request:
The goal of this ticket was to add information on the minimum memory resident ratio for assigning a memory quota to a bucket during bucket creation. 

The Create a Bucket page hadn't been updated in 4 years, so I took the liberty of turning it into a full procedure. It includes the new information about how to assign the memory quota. 

The full explanation of the minimum memory resident ratio is inside the Memory topic. I rewrote the Service Memory Quotas section, added a new Bucket Memory Quotas section, and added to the Ejection section to support the changes to Create a Bucket. 

(I had to do a small rebase because I missed the ticket number in one of my commits :))

sarahlwelton committed:
[DOC-10466] Rewrote create-bucket.adoc to...

...refresh formatting and content. Added information about Memory Quotas.

Updated memory.adoc to add info about Bucket Memory Quotas and...

...Ejection Methods/Policies, to support changes to create-bucket.adoc.

sarahlwelton committed:
Added a Prerequisites section to create-bucket.adoc.

Made changes based on a Vale review of create-bucket.adoc.

Added linking to each section of the Advanced Bucket settings.

Added unordered lists to steps to describe options.

Added information on the Metadata Purge Interval for Ephemeral buckets.

Rewrote the Service Memory Quotas section of memory.adoc for simplicity.

Changed capitalization of Memory Quota in Bucket Memory Quotas section.

Tweaked language and corrected option name in Ejection section.

sarahlwelton committed:
[DOC-10466] Added a Prerequisites section to...

... create-bucket.adoc.

Made changes based on a Vale review of create-bucket.adoc.

Added linking to each section of the Advanced Bucket settings.

Added unordered lists to steps to describe options.

Added information on the Metadata Purge Interval for Ephemeral buckets.

Rewrote the Service Memory Quotas section of memory.adoc for simplicity.

Changed capitalization of Memory Quota in Bucket Memory Quotas section.

Tweaked language and corrected option name in Ejection section.

sarahlwelton committed:
[DOC-10466] Slight language changes to memory.adoc

sarahlwelton committed:
[DOC-10466] Slight language changes to memory.adoc

sarahlwelton committed:
Merge branch 'DOC-10466-minimum-res-ratio' of https://github.com/couchbase/docs-server into DOC-10466-minimum-res-ratio

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,25 @@
-Memory-quota allocation on Couchbase Server occurs _per service_ (except in the cases of the xref:services-and-indexes/services/query-service.adoc[Query Service] and xref:services-and-indexes/services/backup-service.adoc[Backup Service], which do not require specific allocations).
-This allows the availability of memory-resources to be tuned according to the assignment of services, node by node.
-Note that the xref:services-and-indexes/services/data-service.adoc[Data Service] must run on at least one node; and that on each of those nodes, quotas for buckets, specified at the time of bucket-creation, are subtracted from the quota allocated to the Data Service.
+You must configure your memory quota allocations for each service in Couchbase Server, except for the xref:services-and-indexes/services/query-service.adoc[Query Service] and xref:services-and-indexes/services/backup-service.adoc[Backup Service]. You can change the availability of memory resources in your cluster based on how you assign your services to each node.
```
```suggestion
You must configure your memory quota allocations for each service in Couchbase Server, except for the xref:services-and-indexes/services/query-service.adoc[Query Service] and xref:services-and-indexes/services/backup-service.adoc[Backup Service]. 
You can change the availability of memory resources in your cluster based on how you assign your services to each node.
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,25 @@
-The memory-quota allocation specified for a given service applies to every instance of that service across the cluster.
-For example, if 2048 MB is specified for the xref:services-and-indexes/services/analytics-service.adoc[Analytics Service], and the Analytics Service is run on three of a cluster's five nodes, each of the three instances of the service is duly allocated 2048 MB.
-Note that it is not possible to have different memory allocations across multiple instances of the same service within a single cluster.
+The memory quota you allocate for a service applies to every instance of that service across your cluster. For example, if you allocate 2048 MB to the xref:services-and-indexes/services/analytics-service.adoc[Analytics Service], and you run the Analytics Service on three of a cluster's five nodes, each instance of the service has 2048 MB of memory.
```
```suggestion
The memory quota you allocate for a service applies to every instance of that service across your cluster. 
For example, if you allocate 2048 MB to the xref:services-and-indexes/services/analytics-service.adoc[Analytics Service], and you run the Analytics Service on three of a cluster's five nodes, each instance of the service has 2048 MB of memory.
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,25 @@
+The firm limits for server memory allocation can be calculated by: stem:[max(total\_memory - 1, 80\% \times total\_memory)] where `total_memory` is the maximum memory on the node in GiB.
 
-If, when the initial node of a cluster is set up, only a subset of services is assigned, additional nodes can subsequently be added, in order to host additional services: in which case, each new service can be given its initial memory allocation as its node is added.
+The xref:services-and-indexes/services/data-service.adoc[Data Service] must run on at least one node in any cluster. Every xref:buckets.adoc[bucket] you create on a node has its own memory quota. The available memory for a bucket comes from the quota you assign to the Data Service. For more information on bucket memory quotas, see the <<bucket-memory,Bucket Memory Quotas section>>.
```
```suggestion
The xref:services-and-indexes/services/data-service.adoc[Data Service] must run on at least one node in any cluster. Every xref:buckets.adoc[bucket] you create on a node has its own memory quota. 
The available memory for a bucket comes from the quota you assign to the Data Service. 
For more information on bucket memory quotas, see the <<bucket-memory,Bucket Memory Quotas section>>.
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,25 @@
+The xref:services-and-indexes/services/data-service.adoc[Data Service] must run on at least one node in any cluster. Every xref:buckets.adoc[bucket] you create on a node has its own memory quota. The available memory for a bucket comes from the quota you assign to the Data Service. For more information on bucket memory quotas, see the <<bucket-memory,Bucket Memory Quotas section>>.
 
-Instructions on how to add a new node to a cluster can be found in the section xref:manage:manage-nodes/create-cluster.adoc[Create a Cluster].
+When you add a new node, you can choose to use the same configuration from the first node in the cluster. If this default configuration exceeds the memory available for the node, you can customize the new node's settings and change its assigned services. You can add more nodes to a cluster to add more services. 
```
```suggestion
When you add a new node, you can choose to use the same configuration from the first node in the cluster. 
If this default configuration exceeds the memory available for the node, you can customize the new node's settings and change its assigned services. 
You can add more nodes to a cluster to add more services. 
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,25 @@
+For more information about how to add nodes and allocate memory to a service when you initialize a cluster, see xref:manage:manage-nodes/create-cluster.adoc[].
 
-Minimum-required memory-quotas (which are the quota-defaults initially provided by Couchbase Server) are:
+The *Memory Quotas* panel in the xref:manage:manage-settings/general-settings.adoc[General settings screen] of the Couchbase Server Web Console lists all the services running on a cluster. It also shows the memory allocated to each service. You can use the panel to change the memory allocations in your cluster. If a change doesn't meet the required minimum or exceeds the available memory for a cl ...
```
```suggestion
The *Memory Quotas* panel in the xref:manage:manage-settings/general-settings.adoc[General settings screen] of the Couchbase Server Web Console lists all the services running on a cluster. 
It also shows the memory allocated to each service. 
You can use the panel to change the memory allocations in your cluster. 
If a change doesn't meet the required minimum or exceeds the available memory for a cluster, the Memory Quotas panel displays an error. 
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -55,7 +49,23 @@ Minimum-required memory-quotas (which are the quota-defaults initially provided
+[#bucket-memory]
+== Bucket Memory Quotas 
+
+You must specify a memory quota for each bucket you create. This quota is allocated for the bucket on a per node basis and must be less than the total memory quota for the cluster. 
```
```suggestion
You must specify a memory quota for each bucket you create. 
This quota is allocated for the bucket on a per node basis and must be less than the total memory quota for the cluster. 
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -55,7 +49,23 @@ Minimum-required memory-quotas (which are the quota-defaults initially provided
+
+You must specify a memory quota for each bucket you create. This quota is allocated for the bucket on a per node basis and must be less than the total memory quota for the cluster. 
+
+Set the memory quota based on the expected size of your dataset. The memory quota for a bucket must support the minimum memory resident ratio of its xref:learn:buckets-memory-and-storage/storage-engines.adoc[storage engine]: 
```
```suggestion
Set the memory quota based on the expected size of your dataset. 
The memory quota for a bucket must support the minimum memory resident ratio of its xref:learn:buckets-memory-and-storage/storage-engines.adoc[storage engine]: 
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -55,7 +49,23 @@ Minimum-required memory-quotas (which are the quota-defaults initially provided
+
+For example, if you expect to have about 2TBs of data per node in your cluster and want to use the *Magma* engine, you could set the memory quota for a bucket to 40GB. 
+
+NOTE: These values are recommendations only. The specific memory quota requirements for your bucket are dependent on access patterns, data density, and other factors.
```
```suggestion
NOTE: These values are recommendations only. 
The specific memory quota requirements for your bucket are dependent on access patterns, data density, and other factors.
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -88,9 +98,22 @@ See xref:cli:cbepctl/set-flush_param.adoc[set flush_param].
-Note that in some cases, ejection is configured _not_ to occur.
-For detailed information, see xref:buckets-memory-and-storage/buckets.adoc[Buckets].
+
+Different ejection methods are available, and are configured per bucket. Note that in some cases, ejection is configured _not_ to occur.
```
```suggestion
Different ejection methods are available, and are configured per bucket. 
Note that in some cases, ejection is configured _not_ to occur.
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -88,9 +98,22 @@ See xref:cli:cbepctl/set-flush_param.adoc[set flush_param].
+
+For a Couchbase bucket, you can choose betweeen a *Value-only* or *Full* ejection method: 
+
+* *Value-only*: The bucket only ejects data when it removes a document from memory. Choose this method if you need better performance, but be aware that it uses more system memory. 
```
```suggestion
* *Value-only*: The bucket only ejects data when it removes a document from memory. 
Choose this method if you need better performance, but be aware that it uses more system memory. 
```

RayOffiah commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -88,9 +98,22 @@ See xref:cli:cbepctl/set-flush_param.adoc[set flush_param].
+For a Couchbase bucket, you can choose betweeen a *Value-only* or *Full* ejection method: 
+
+* *Value-only*: The bucket only ejects data when it removes a document from memory. Choose this method if you need better performance, but be aware that it uses more system memory. 
+* *Full*: The bucket ejects data, metadata, keys, and values when it removes a document from memory. Choose this method if you want to reduce your memory overhead requirement. 
```
```suggestion
* *Full*: The bucket ejects data, metadata, keys, and values when it removes a document from memory. 
Choose this method if you want to reduce your memory overhead requirement. 
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,246 @@
 
-[#add-data-bucket-dialog-initial]
-image::manage-buckets/addDataBucketDialogInitial.png[,320,align=left]
+NOTE: Your memory quota needs to match the minimum memory resident ratio required by your chosen storage engine. For more information, see xref:learn:buckets-memory-and-storage/memory.adoc#bucket-memory[Bucket Memory Quotas].
```
```suggestion
NOTE: Your memory quota needs to match the minimum memory resident ratio required by your chosen storage engine. 
For more information, see xref:learn:buckets-memory-and-storage/memory.adoc#bucket-memory[Bucket Memory Quotas].
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,246 @@
+
+.. In the *Seconds* field, enter the maximum time in seconds that a document can exist in the bucket before it's deleted. 
++
+TIP: The maximum allowed value is 2147483648 seconds (68.096 years). You can only apply this setting to documents created after you change the configuration. 
```
```suggestion
TIP: The maximum allowed value is 2147483648 seconds (68.096 years). 
You can only apply this setting to documents created after you change the configuration. 
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,246 @@
-Note that you can also set the conflict resolution method using the CLI xref:cli:cbcli/couchbase-cli-bucket-create.adoc[bucket-create] command, or the xref:rest-api:rest-bucket-create.adoc[REST API].
+Bucket Priority sets the priority of the bucket's background tasks relative to the background tasks of other buckets on the cluster. 
++
+Background tasks may involve disk I/O, DCP stream-processing, item-paging, and more. Specifying High might result in faster processing for the current bucket's tasks. This setting only takes effect when there is more than one bucket defined for the cluster, and you have assigned different Bucket Priority values.
```
```suggestion
Background tasks may involve disk I/O, DCP stream-processing, item-paging, and more. 
Specifying High might result in faster processing for the current bucket's tasks. 
This setting only takes effect when there is more than one bucket defined for the cluster, and you have assigned different Bucket Priority values.
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,246 @@
-Note that _ejection_ when applied to an Ephemeral bucket means removal of bucket-data from memory without persistence (since ephemeral buckets have no presence on disk).
+.. In the *Seconds* field, enter the maximum number of seconds a document can exist in the bucket before it's deleted. 
++
+TIP: The maximum allowed value is 2147483648 seconds (68.096 years). You can only apply this setting to documents created after you change the configuration. 
```
```suggestion
TIP: The maximum allowed value is 2147483648 seconds (68.096 years). 
You can only apply this setting to documents created after you change the configuration. 
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,246 @@
++
+Bucket Priority sets the priority of the bucket's background tasks relative to the background tasks of other buckets on the cluster. 
++
+Background tasks may involve DCP stream-processing, item-paging, and more. Specifying High might result in faster processing for the current bucket's tasks. This setting only takes effect when there is more than one bucket defined for the cluster, and the buckets are assigned different Bucket Priority values.
```
```suggestion
Background tasks may involve DCP stream-processing, item-paging, and more. 
Specifying High might result in faster processing for the current bucket's tasks. T
his setting only takes effect when there is more than one bucket defined for the cluster, and the buckets are assigned different Bucket Priority values.
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,246 @@
-image::manage-buckets/bucketsViewWithCreatedBucket.png[,750,align=left]
+. In the *Metadata Purge Interval* field, enter a value between `0.0007-60` to set how often a node purges metadata on deleted items.
++
+A value of `0.0007` equals a minute. A value of `0.5` equals 12 hours. If this value is too high, the node might have a delay when reclaiming memory. If set too low, data might be inconsistent in XDCR or Views. 
```
```suggestion
A value of `0.0007` equals a minute. 
A value of `0.5` equals 12 hours. 
If this value is too high, the node might have a delay when reclaiming memory. 
If set too low, data might be inconsistent in XDCR or Views. 
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -202,17 +256,15 @@ For example:
-Note that a minimum durability level of `persistToMajority` is specified: other permitted settings for this parameter are `none` (the default), `majorityAndPersistActive`, and `majority`.
-For details on the significance of these settings, see xref:learn:data/durability.adoc[Durability].
-(Note that an _Ephemeral_ bucket, when created, can only use the `none` and `majority` settings, for this parameter.)
+The preceding example creates a `Couchbase` bucket named `testBucket`, with a RAM size of `1024`. It sets a Maximum Time-to-Live and disables Flush. It also sets a Minimum Durability Level of `persistToMajority`. 
```
```suggestion
The preceding example creates a `Couchbase` bucket named `testBucket`, with a RAM size of `1024`.
 It sets a Maximum Time-to-Live and disables Flush. 
 It also sets a Minimum Durability Level of `persistToMajority`. 
```

RayOffiah commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -224,9 +276,6 @@ curl -v -X POST http://10.143.201.101:8091/pools/default/buckets \
-Note that a minimum durability level of `majorityAndPersistActive` is specified: other permitted settings for this parameter are `none` (the default), `majority`, and `persistToMajority`.
-For details on the significance of these settings, see xref:learn:data/durability.adoc[Durability].
-(Note that an _Ephemeral_ bucket, when created, can only use the `none` and `majority` settings, for this parameter.)
+The preceding example creates a `Couchbase` bucket named `testBucket`, with a RAM size of `512`. It sets a Minimum Durability Level of `majorityAndPersistActive`.
```
```suggestion
The preceding example creates a `Couchbase` bucket named `testBucket`, with a RAM size of `512`. 
It sets a Minimum Durability Level of `majorityAndPersistActive`.
```

RayOffiah reviewed (changes requested):
Text is fine, but we aim for one sentence per line. 👍🏾

sarahlwelton committed:
Apply suggestions from code review

Co-authored-by: Ray <77050471+RayOffiah@users.noreply.github.com>

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Ephemeral*
++
+For more information about bucket types, see xref:learn:buckets-memory-and-storage/buckets.adoc[].
+. Choose a *Storage Backend* for the bucket:
```
The paragraph beginning "For more information..." belongs semantically to the numbered step, not to the last bullet point. I think it should be outdented so that it aligns with the text beginning "Choose a **Bucket Type** ..."
To do this, you need to add a single blank line before the continuation mark `+`:

```suggestion
* *Ephemeral*

+
For more information about bucket types, see xref:learn:buckets-memory-and-storage/buckets.adoc[].
. Choose a *Storage Backend* for the bucket:
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Couchstore*
+* *Magma*
++
+For more information about the available storage engines, see xref:learn:buckets-memory-and-storage/storage-engines.adoc[Storage Engines].
```
Likewise, outdent the "For more information..." paragraph slightly to line it up with "Choose a **Storage Backend** ..."

```suggestion
* *Magma*

+
For more information about the available storage engines, see xref:learn:buckets-memory-and-storage/storage-engines.adoc[Storage Engines].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
-The [.ui]*Buckets* screen now appears, showing the buckets that have already been defined for your system.
-If you are accessing this screen immediately after setting up the cluster, no buckets are yet defined:
+[#add-data-bucket-dialog-initial]
+image::manage-buckets/addDataBucketDialogInitial.png[,320,align=left]
```
I know you inherited this, but I think screenshots should be center-aligned throughout. Left-aligned screenshots end up  outside the left margin of the text in a multi-level bulleted / numbered list, which visually breaks up the procedure.

(There's a case that other types of images should be left-aligned in other contexts.)

Also consider adding alt text for all screenshots if you have time &mdash; for example, "The Add Data Bucket dialog with initial options"

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+. Select btn:[Add Bucket].
 
-Specify the [.ui]*Storage Backend* for the bucket, which can be `Couchstore` or `Magma`.
+The bucket appears on the *Buckets* screen: 
```
I recommend not using a colon before a screenshot. If the screenshot is deleted in a future revision, or the paragraph gets moved around, the colon is just one extra thing to remove and it frequently gets left.

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
 
-[#add-data-bucket-dialog-expanded]
-image::manage-buckets/addBucketWithMagmaOption.png[,400,align=left]
+. To enable xref:learn:clusters-and-availability/intra-cluster-replication.adoc[replica creation and management], under *Replicas*, select the btn:[Enable] checkbox. 
```
I don't think a check box is a button. It's not something the user clicks to make an action happen straight away. I would just use bold for check boxes throughout.

sarahlwelton replied:
Yeah I waffled on this a bit just because of the [.ui] labels that were left on all of the checkboxes.

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+
+.. In the *Seconds* field, enter the maximum time in seconds that a document can exist in the bucket before it's deleted. 
++
+TIP: The maximum allowed value is 2147483648 seconds (68.096 years). 
```
This admonition is part of the substep above (so appears indented) &mdash; which is fair enough, but is inconsistent with the "memory quota" note on line 56, which is outside the flow of the numbered steps. Needs to be consistent either way.

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* The number of items the bucket contains. 
+* The percentage of items in the bucket that are currently resident. 
+* The number of operations per second being performed on the bucket. 
+* The amount of RAM the bucekt is currently using, compared to its memory quota. 
```
Typo

```suggestion
* The amount of RAM the bucket is currently using, compared to its memory quota. 
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
-By checking the [.ui]*Replicate view indexes* checkbox, you ensure that view indexes, as well as data, are replicated.
+* *Off* 
+* *Passive*
+* *Active*
```
Add a single blank line before the continuation symbol `+`, so that "For more information ..." lines up with "Choose a Compression mode ..."

```suggestion
* *Active*

```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Sequence number* 
+* *Timestamp*
++
+For more information about XDCR conflict resolution, see xref:learn:clusters-and-availability/xdcr-conflict-resolution.adoc[].
```
Outdent "For more information ...." to line up with "Choose a ..."

```suggestion
* *Timestamp*

+
For more information about XDCR conflict resolution, see xref:learn:clusters-and-availability/xdcr-conflict-resolution.adoc[].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Value-only*
+* *Full*
++
+For more information about ejection, see the xref:learn:buckets-memory-and-storage/memory.adoc#ejection[Ejection section in Memory].
```
Outdent as above

```suggestion
* *Full*

+
For more information about ejection, see the xref:learn:buckets-memory-and-storage/memory.adoc#ejection[Ejection section in Memory].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
++
+* *Default*
+* *High*
 +
```
Outdent as above

```suggestion
* *High*

+
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *majorityAndPersistActive*
+* *persistToMajority*
++
+For more information about durability, see xref:learn:data/durability.adoc[].
```
Outdent as above

```suggestion
* *persistToMajority*

+
For more information about durability, see xref:learn:data/durability.adoc[].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+.. In the *Number of replica (backup) copies* list, select the number of replicas for the bucket. 
 
-The following settings are different for Ephemeral buckets:
+. . To set a xref:learn:data/expiration.adoc[document expiration] for documents in the bucket, under *Bucket Max Time-To-Live*, select the btn:[Enable] checkbox.
```
Extra full stop?

```suggestion
. To set a xref:learn:data/expiration.adoc[document expiration] for documents in the bucket, under *Bucket Max Time-To-Live*, select the btn:[Enable] checkbox.
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Passive*
+* *Active* 
++
+For more information about the available compression modes, see xref:learn:buckets-memory-and-storage/compression.adoc[].
```
Outdent as above

```suggestion
* *Active* 

+
For more information about the available compression modes, see xref:learn:buckets-memory-and-storage/compression.adoc[].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Sequence number*
+* *Timestamp* 
++
+For more information about XDCR conflict resolution, see xref:learn:clusters-and-availability/xdcr-conflict-resolution.adoc[].
```
Outdent as above

```suggestion
* *Timestamp* 

+
For more information about XDCR conflict resolution, see xref:learn:clusters-and-availability/xdcr-conflict-resolution.adoc[].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *Default*
+* *High*
++
+Bucket Priority sets the priority of the bucket's background tasks relative to the background tasks of other buckets on the cluster. 
```
Outdent as above

```suggestion
* *High*

+
Bucket Priority sets the priority of the bucket's background tasks relative to the background tasks of other buckets on the cluster. 
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
++
+Background tasks may involve DCP stream-processing, item-paging, and more. 
+Specifying High might result in faster processing for the current bucket's tasks. T
+his setting only takes effect when there is more than one bucket defined for the cluster, and the buckets are assigned different Bucket Priority values.
```
Typo

```suggestion
Specifying High might result in faster processing for the current bucket's tasks.
This setting only takes effect when there is more than one bucket defined for the cluster, and the buckets are assigned different Bucket Priority values.
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *No ejection* 
+* *Eject data when RAM is full* 
++
+For more information about ejection, see the xref:learn:buckets-memory-and-storage/memory.adoc#ejection[Ejection section in Memory].
```
Outdent as above

```suggestion
* *Eject data when RAM is full* 

+
For more information about ejection, see the xref:learn:buckets-memory-and-storage/memory.adoc#ejection[Ejection section in Memory].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* *none*
+* *majority*
++
+For more information about durability, see xref:learn:data/durability.adoc[].
```
Outdent as above

```suggestion
* *majority*

+
For more information about durability, see xref:learn:data/durability.adoc[].
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+
 For example:
 
 ----
```
Markup as code so users can copy and paste

```suggestion
[source,sh]
----
```

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -202,17 +266,17 @@ For example:
+
 For example:
 
 ----
```
Markup as code so users can copy and paste

```suggestion
[source,sh]
----
```

simon-dew commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,33 @@
 
-When a node is added to a cluster, the _Default Configuration_, as established by the set-up of the first node in the cluster, is available: this covers all configurable elements, including memory quotas.
-However, if insufficient memory for the default configuration is available on the new node, the default configuration is prohibited: in such cases, settings for the new node can be custom-configured, allowing an appropriate subset of services to be specified.
+The firm limits for server memory allocation can be calculated by: stem:[max(total\_memory - 1, 80\% \times total\_memory)] where `total_memory` is the maximum memory on the node in GiB.
```
The STEM formula breaks the line, leaving an ugly gap at the end of the previous line. Is there any way to shorten the preceding text to avoid the line break? Or put the formula in a line by itself?

sarahlwelton replied:
I like line by itself.

simon-dew commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,33 @@
+For more information on bucket memory quotas, see the <<bucket-memory,Bucket Memory Quotas section>>.
 
-Instructions on how to add a new node to a cluster can be found in the section xref:manage:manage-nodes/create-cluster.adoc[Create a Cluster].
+When you add a new node, you can choose to use the same configuration from the first node in the cluster. 
+If this default configuration exceeds the memory available for the node, you can customize the new node's settings and change its assigned services. 
+You can add more nodes to a cluster to add more services. 
```
This is true, but some administrators deliberately set up different services on different nodes. So when they add a new node, they will as you say "customize the new node's settings and change its assigned services", but it won't be because "the default configuration exceeds the memory available for the node"; it's because they _don't want to use_ the default configuration on the new node. Need to rewrite to reflect this.

simon-dew commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,33 @@
-If, when the initial node of a cluster is set up, only a subset of services is assigned, additional nodes can subsequently be added, in order to host additional services: in which case, each new service can be given its initial memory allocation as its node is added.
+The xref:services-and-indexes/services/data-service.adoc[Data Service] must run on at least one node in any cluster. Every xref:buckets.adoc[bucket] you create on a node has its own memory quota. 
+The available memory for a bucket comes from the quota you assign to the Data Service. 
+For more information on bucket memory quotas, see the <<bucket-memory,Bucket Memory Quotas section>>.
```
Not a big deal, but we don't tend to use the word "section" within the link text. Outside the link text is fine.

simon-dew reviewed (changes requested):
This is fantastic. The two pages are much clearer and easier to read than before.

Sorry for all the comments, it looks like a lot, but they're small and repetitive. The only general point I'd add is to echo Ray's comment on [ventilated prose](https://writetheasciidocs.netlify.app/ventilated-prose): new line for each sentence.

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,192 +1,256 @@
+* You must be a Full or Cluster Administrator.
 
 [#create-bucket-using-couchbase-web-console]
 == Create a Bucket with The UI
```
"The" shouldn't be capitalized here

```suggestion
== Create a Bucket with the UI
```

sarahlwelton committed:
[DOC-10466] Adding changes from peer review

sarahlwelton committed:
[DOC-10466] Removing missed btn macro from...

create-bucket.adoc.

Indenting a note block to match the surrounding text.

sarahlwelton committed:
[DOC-10466] Fixing indenting on steps in...

...create-bucket.adoc.

Moved word "section" out of link text on one instruction, for consistency.

simon-dew commented on modules/manage/pages/manage-buckets/create-bucket.adoc:
```diff
@@ -1,194 +1,272 @@
 
-The row for the created bucket provides data on the number of items the bucket contains and how many are currently resident; on how many operations per second are being performed on the bucket; on the amount of RAM currently being used by the bucket, in comparison to its maximum quota; and the amount of disk space used by the bucket. To the right hand side of the row are tabs that allow examinati ...
+Background tasks may involve DCP stream-processing, item-paging, and more. 
+Specifying High might result in faster processing for the current bucket's tasks. T
```
The typo "T" is still there at the end of the line

```suggestion
Specifying High might result in faster processing for the current bucket's tasks.
```

simon-dew commented on modules/learn/pages/buckets-memory-and-storage/memory.adoc:
```diff
@@ -9,31 +9,37 @@
-Minimum-required memory-quotas (which are the quota-defaults initially provided by Couchbase Server) are:
+When you add a new node, you can use the same configuration and services from the first node in the cluster. 
+You can also choose to customize the new node's settings and change its assigned services. 
+You can add more nodes to a cluster to add more services. 
```
The first sentence sums everything up by itself; don't think we need the second one here.

```suggestion
You can also choose to customize the new node's settings and change its assigned services.
```

simon-dew reviewed (commented):
Thanks @sarahlwelton &mdash; just two comments this time.

sarahlwelton committed:
Update modules/manage/pages/manage-buckets/create-bucket.adoc

Co-authored-by: Simon Dew <39966290+simon-dew@users.noreply.github.com>

sarahlwelton committed:
Update modules/learn/pages/buckets-memory-and-storage/memory.adoc

Co-authored-by: Simon Dew <39966290+simon-dew@users.noreply.github.com>

simon-dew reviewed (approved):
LGTM 👍
